### ISSUE TITLE
Cache Dep details in the Dep object

### ISSUE THREAD

**Issue by dave-2** [catapult-project/catapult#3822]
### subject, author, timestamp
Right now, to show Dep details (subject, author, timestamp) in the UI, Pinpoint does an XHR call to Gitiles. This leads to noticeable loading times when clicking on points on the graph. Since Pinpoint already does these lookups in the backend when creating and running a job, we might as well store this information in the Dep in the datastore.

### commit URLs
It's also very slow to look up the commit URLs, because `namespaced_stored_object.Get(_REPOSITORIES_KEY)` is called for each Dep, and it's a slow call. Because this is done on the backend side, the entire UI is slow. I propose caching this too. One caveat is that the repository URLs can change over time -- that's why we don't already store them. We can mitigate this somewhat by only using this cached object for the UI (in `AsDict()`) and doing the lookup for the backend.

### Migration
So the new fields are `subject`, `author`, `timestamp`, `url`.

For migration, I'm going to create a `Commit` object that contains the new fields, and as Jobs are loaded, convert their `Dep` objects to `Commit` objects. I think `Commit` makes more sense as the object name than `Dep` anyway.

@anniesullie @perezju @simonhatch

**Comment by perezju** [comment]
+1 to `Dep` → `Commit` and caching subject, author, timestamp.

On commit URLs, would it be better to just cache the value of `namespaced_stored_object.Get(_REPOSITORIES_KEY)` so a single call to it is needed for the lifetime of each backend instance?

**Comment by dave-2** [comment]
That sounds like a good solution. We can just stuff it in a module-level variable.

**Comment by dave-2** [comment]
So what happened is I tried to do that, but quickly learned that it wouldn't work for unit tests. We would need to clear the cache in the `tearDown()` of every test. Fine, not the best, but it works.

But then the tests started failing randomly. Turns out that they now have race conditions, because they run in parallel. I then tried to use thread-local storage, but its behavior is super weird.

tl;dr: #3834 is going to solve the repository URL caching.

**Comment by dave-2** [comment]
Old issue, #3834 and #4114 have discussions about the newer work done / being done here.
